apps/epic-web/app/routes/index.tsx

```diff
@@ -1,7 +1,88 @@
+import type {ActionFunction, LoaderFunction} from '@remix-run/node'
+import {redirect, json} from '@remix-run/node'
+import {z} from 'zod'
+import {subscribeToForm} from '../lib/convertkit.server'
+import {Form, useLoaderData} from '@remix-run/react'
+import {convertkitSubscriberCookie} from '~/cookies'
+
+const FORM_ID = '3547851'
+
+export const loader: LoaderFunction = async ({request}) => {
+  const cookieHeader = request.headers.get('Cookie')
+  const ckSubscriber =
+    (await convertkitSubscriberCookie.parse(cookieHeader)) || {}
+
+  return json({ckSubscriber})
+}
+
+export const action: ActionFunction = async ({request}) => {
+  const formPayload = Object.fromEntries(await request.formData())
+
+  const validationSchema = z.object({
+    first_name: z.string(),
+    email: z.string().email(),
+  })
+
+  try {
+    const validatedSchema = validationSchema.parse(formPayload)
```

Comment:
zod takes your random object, runs it against the schema, and if it is the thing you get a nice typed object back to utilize